### ISSUE TITLE
library loading error

### ISSUE THREAD

**Issue by bekker** [neutrinolabs/xrdp#689]
0.9.1 master branch
I configured xrdp with `--enable-painter` option, then made it and installed it.
Added library path in `/etc/ld.so.conf.d`, updated ldconfig cache.

I get this error in the log after authentication, and can't enter the desktop.
```
(7083)(140349989779200)[ERROR] xrdp_wm_log_msg: error loading libxup.so specified in xrdp.ini, please add a valid entry like lib=libxrdp-vnc.so or similar
```

Output of `ldconfig -p | grep libxup`
```
libxup.so.0 (libc6,x86-64) => /usr/local/lib/xrdp/libxup.so.0
libxup.so (libc6,x86-64) => /usr/local/lib/xrdp/libxup.so
```

Same for vnc. Setting `LD_LIBRARY_PATH` nor rebooting didn't work.
The desktop environment is mate (`mate-session` in `~/.xsession`)

It seems to be my mistake, but I couldn't find solutions anywhere.
Did I miss something or maybe a bug?

**Comment by bekker** [comment]
```
$ ldd /usr/local/lib/xrdp/libxup.so
linux-vdso.so.1 =>  (0x00007ffeb5deb000)
libcommon.so.0 => /usr/local/lib/xrdp/libcommon.so.0 (0x00007fbb8032c000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fbb7ff2e000)
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fbb7fd10000)
libssl.so.1.0.0 => /lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007fbb7faa7000)
libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007fbb7f663000)
libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fbb7f45e000)
/lib64/ld-linux-x86-64.so.2 (0x0000555b0f1ac000)
```

**Comment by bekker** [comment]
Reinstalling `xorgxrdp` fixed the issue. I don't know why, but it works perfect.

**Comment by proski** [comment]
`libxup.so` is a loadable module, not a library. It should not be in `/etc/ld.so.conf.d` or in `LD_LIBRARY_PATH`. xrdp loads modules on its own, and it knows the directory. If any documentation make you think that the modules should be in the library path, please provide a link. The documentation may need to be clarified.

I cannot reproduce the issue with the master branch. I added an invalid call to xup.c, and now xrdp prints the unresolved symbol to the console (it needs to be run with `--no-daemon` for that)

```
xrdp: symbol lookup error: /usr/local/lib/xrdp/libxup.so: undefined symbol: foo_bar
```

I don't see the message you quoted. I suspect `libxup.so` wasn't installed at all. Maybe "make install" failed or is was run in a subdirectory of xrdp.

**Comment by bekker** [comment]
@proski Oh then it's my mistake. At first I tried to install with `X11RDP-o-Matic`and found that it doesn't support 0.9.1 yet. So I moved into xrdp directory in it and installed myself. I figured it's the same repo, so everything would be same.

The real problem was that it was configured by `X11RDP-o-matic` and I myself installed it in the shell. So some kind of configuration must have gone wrong. xrdp tries to load cursors and logo from `/usr/...` not `/usr/local/...`, whereas they were really in `/usr/local/...`. I made symlinks for those assets, but not libraries.

I saw the messages from `libtool` when installing that I should `ldconfig` those libraries... It was false warning after all. Thank you for the reply.
